mistralrs-core/src/lib.rs

```diff
@@ -363,48 +363,53 @@ impl MistralRs {
 
         let engine_id = ENGINE_ID.fetch_add(1, atomic::Ordering::SeqCst);
 
-        // Do a dummy run
-        if matches!(category, ModelCategory::Text | ModelCategory::Vision { .. }) {
-            let clone_sender = sender.read().unwrap().clone();
-            tokio::task::block_in_place(|| {
-                let (tx, mut rx) = channel(1);
-                let req = Request::Normal(NormalRequest {
-                    id: 0,
-                    messages: RequestMessage::Completion {
-                        text: "dummy".to_string(),
-                        echo_prompt: false,
-                        best_of: None,
-                    },
-                    sampling_params: SamplingParams {
-                        max_len: Some(1),
-                        ..SamplingParams::deterministic()
-                    },
-                    response: tx,
-                    return_logprobs: false,
-                    is_streaming: true,
-                    constraint: Constraint::None,
-                    suffix: None,
-                    adapters: None,
-                    tool_choice: None,
-                    tools: None,
-                    logits_processors: None,
-                    return_raw_logits: false,
+        // Skip the dummy run in Single Threaded mode as blocking operations are not allowed
+        if tokio::runtime::Handle::try_current()
+            .is_ok_and(|h| h.runtime_flavor() != tokio::runtime::RuntimeFlavor::CurrentThread)
+        {
+            // Do a dummy run
+            if matches!(category, ModelCategory::Text | ModelCategory::Vision { .. }) {
+                let clone_sender = sender.read().unwrap().clone();
+                tokio::task::block_in_place(|| {
+                    let (tx, mut rx) = channel(1);
+                    let req = Request::Normal(NormalRequest {
+                        id: 0,
+                        messages: RequestMessage::Completion {
+                            text: "dummy".to_string(),
+                            echo_prompt: false,
+                            best_of: None,
+                        },
+                        sampling_params: SamplingParams {
+                            max_len: Some(1),
+                            ..SamplingParams::deterministic()
+                        },
+                        response: tx,
+                        return_logprobs: false,
+                        is_streaming: true,
+                        constraint: Constraint::None,
+                        suffix: None,
+                        adapters: None,
+                        tool_choice: None,
+                        tools: None,
+                        logits_processors: None,
+                        return_raw_logits: false,
+                    });
+                    info!("Beginning dummy run.");
+                    let start = Instant::now();
+                    clone_sender.blocking_send(req).unwrap();
```

Comment:
Wouldn't it be better to just do `clone_sender.send(req)` and the `recv()` equivalent?

Reply:
send and recv requires async context so can't be used here unfortunatelly